wgpu-hal/src/vulkan/instance.rs

```diff
@@ -874,22 +850,19 @@ impl crate::Instance for super::Instance {
             (Rwh::AndroidNdk(handle), _) => {
                 self.create_surface_android(handle.a_native_window.as_ptr())
             }
-            #[cfg(windows)]
             (Rwh::Win32(handle), _) => {
-                use winapi::um::libloaderapi::GetModuleHandleW;
-
-                let hinstance = unsafe { GetModuleHandleW(std::ptr::null()) };
-                self.create_surface_from_hwnd(hinstance as *mut _, handle.hwnd.get() as *mut _)
+                let hinstance = handle.hinstance.expect("HINSTANCE is not set.");
+                self.create_surface_from_hwnd(hinstance.get(), handle.hwnd.get())
```

Comment:
Any reason to call the Windows function when the `raw-window-handle` contains `hinstance` already? Probably and oversight since, as history suggests, when this code was introduced in 2021 on `raw-window-handle` the field was already there.

Reply:
Yeah, I'm not actually certain why and in what case it is supposed or allowed to be left at zero. Nor if it makes sense for `wgpu` to fall back to `GetModuleHandleW()` if it's unset.

I've never seen it being `None`/`0`, at least not with `winit`.

[Vulkan spec demands it to be valid](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkWin32SurfaceCreateInfoKHR.html), and I've converted the unwrap into returning a proper `InstanceError`: should we do the same to the `expect()`s at `Xlib` and `Xcb`?

Reply:
I'm not sure, are these errors recoverable? I would generally lean towards errors for libraries; although in this case `InstanceError` is opaque... 😅

Reply:
I think we should keep things consistent though.

@cwfitzgerald do you have an opinion on panic vs error for these?

Reply:
Not sure either, I'm not a windows expert on this front. We could keep the `.expect()` and ask the `raw-window-handle` devs what made them consider this optional?

Seems like there are multiple ways to get this opaque handle anyway, via `GetModuleHandleW(NULL)` for the current module or [`GetWindowLongPtrA(hwnd, GWLP_HINSTANCE)`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getwindowlongptra)...

Reply:
InstanceError is fine here I think.